ui/src/server/handlers/api.js

```diff
@@ -1514,24 +1514,36 @@ Fetchr.registerService({
                 break;
             }
             case 'policy': {
-                req.clients.zms.putPolicyVersion(
+                req.clients.zms.putPolicy(
                     {
                         domainName: params.domainName,
                         policyName: params.collectionName,
                         policy: params.detail,
                         auditRef: params.auditRef,
+                        returnObj: true,
                     },
                     responseHandler.bind({ caller: 'putPolicy', callback, req })
                 );
                 break;
             }
             case 'service': {
+                if (params.detail) {
+                    // Note: zms expects publicKeys to be an array but we store it as an object
+                    // Additionally, updating service tags should not modify existing public keys
+                    let publicKeysList = [];
+                    let publicKeysMap = params.detail.publicKeys || {};
+                    for (const [, value] of Object.entries(publicKeysMap)) {
+                        publicKeysList.push(value);
+                    }
+                    params.detail.publicKeys = publicKeysList;
+                }
```

Comment:
TODO: 
We currently have many instances where we deal with publicKeys expecting them to be objects instead of arrays. https://github.com/AthenZ/athenz/blob/master/ui/src/redux/reducers/services.js#L120-L132 I'm proposing we address this in a follow up pr as a full fix is quite involved (and dangerous so will need lots of testing).

Reply:
ticket created - https://ouryahoo.atlassian.net/browse/ATHENS-7444